pytorch_lightning/trainer/trainer.py

```diff
@@ -2094,10 +2094,17 @@ def _results(self) -> Optional[ResultCollection]:
             return active_loop._results
 
     def _exit_gracefully_on_signal(self) -> None:
-        if _fault_tolerant_training() and self._terminate_gracefully:
-            caller = inspect.stack()[1]
-            class_name = caller[0].f_locals["self"].__class__.__name__
-            raise ExitGracefullyException(f"Exiting gracefully on {class_name}:{caller.function}")
+        if not _fault_tolerant_training():
+            return
+        if not self._should_terminated_gracefully():
+            return
```

Comment:
```suggestion
        if not _fault_tolerant_training() or not self._should_terminated_gracefully():
            return
```

Reply:
No, you don't want to force a reduce when fault tolerant is not active. This would be costly for the users who don't want to opt-in.